### PR TITLE
Update linting rules to catch a range of Python issues

### PR DESCRIPTION
**Pull request by dhilton**
### What is the context of this PR?

This PR adds extra linting rules to our CI setup to help us catch common errors or causes of bugs in the way in which we write python code.  It is influenced by some of the approaches taken by Mozilla: https://bugzilla.mozilla.org/show_bug.cgi?id=1237977
### How to review
1. Check out the branch
2. Run `pip install -r requirement.txt` to install new flake8 plugins
3. Run `./scripts/run_tests.sh` to execute the Flake8 code linting
4. Check that they all pass.
- [ ] Do all commits have sensible commit messages?
- [ ] Is all new code unit tested?
- [ ] Are there integration tests if needed?
- [ ] Is there sufficient logging?
- [ ] Is there documentation or is the code self-describing?
### Who worked on the PR

@dhilton

### CODE REVIEW

**Comment by dhilton** [comment]
Looks good - can we just rebase and amend the commit messages and then this is ready to go?

**Comment by dhilton** [comment]
After discussions, let's merge #362 and @LJBabbage 's work first then rebase and fix issues with those code changes.

File: app/templating/template_pre_processor.py [hunk 1]
```diff
@@ -0,0 +1,201 @@
+import logging
```
Comment (warrenbailey) [inline app/templating/template_pre_processor.py]: There's something wrong here, this file shouldn't exist anymore?

**Comment by warrenbailey** [comment]
That template preprocessor file shouldn't be in this commit, it was deleted earlier this week. Sounds like a git rebase issue.

**Comment by warrenbailey** [comment]
Please can we squash and tidy up the commits?

**Comment by warrenbailey** [comment]
LGTM 👍

**Pull request merged**